spec/support/helpers/shared_contexts.rb

```diff
@@ -0,0 +1,4 @@
+RSpec.shared_context 'generators', type: :generator do
+  destination File.expand_path('../../../tmp', __FILE__)
+  before { prepare_destination }
+end
```

Comment:
At first I was thinking this file should be called `generator_shared_context.rb` since the current name suggests that all future shared contexts should live inside this file, which I'd like to avoid. But looking through the other helper files, I realized that this is actually the first shared context. So what do you think of moving the file to a new directory: `spec/support/shared_contexts/generator.rb`. We could then consider _not_ adding `spec/support/shared_contexts` to the list of magically required file and start following your suggestion to require support files explicitly. In separate PRs we could start doing the same for the other sub directories of `support`.

Sorry, for suggesting to undo my last change request, but I really think your argument for explicitness makes a lot of sense and a new support sub directory gives us a good opportunity to start moving in that direction without giving up the current structure of `spec/support`.

Other than that, everything looks really great! Please squash the commits and I'll merge.